src/ui/home/containers/claimDetails.tsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react'
+import { ClaimDetailsComponent } from 'src/ui/home/components/claimDetails'
+import { connect } from 'react-redux'
+import { RootState } from 'src/reducers/'
+import { accountActions } from 'src/actions'
+import { DecoratedClaims } from 'src/reducers/account/'
+import { ClaimsState } from 'src/reducers/account'
+import Immutable from 'immutable'
+
+interface ConnectProps {
+  claims: ClaimsState
+  saveClaim: (claimsItem: DecoratedClaims) => void
+}
+
+interface Props extends ConnectProps {}
+
+interface State {
+  selectedClaim: DecoratedClaims
+}
+
+export class ClaimDetailsContainer extends React.Component<Props, State> {
+  state = {
+    selectedClaim: {
+      displayName: '',
+      type: ['', ''],
+      claims: [],
+    }
+  }
+
+  componentWillMount() {
```

Comment:
I think we can remove this logic since we have the selected claim already in the state; no need to filter it from the overall categorized claims object.